misk/src/main/kotlin/misk/client/TypedHttpClientModule.kt

```diff
@@ -89,5 +91,16 @@ class TypedHttpClientModule<T : Any>(
           invocationHandler
       ) as T
     }
+
+    private fun getConverter(endpointConfig: HttpClientEndpointConfig): Converter.Factory {
+      return when {
+        endpointConfig.converter == "json" -> MoshiConverterFactory.create(moshi)
```

Comment:
could we just inspect the retrofit interface and determine if the request/response types are wire messages?

Reply:
I didn't mean the implementations, but the actual request/response type arguments. e.g `ReturnADinosaur` extends `Message`

Reply:
Ah, sure, I think you could do that, but that's kind of a tricky game. For example, would a string by plaintext or json? Retrofit allows you to actually specify the content-type headers and I think that's probably the right way to do it. Then you can just look at that and set the converter accordingly and default to something sane (e.g. json), if not present.

Reply:
a string is json, since it's not a Message.

but we're not using the content-type headers. we're statically defining it

Reply:
You can statically define the content-type headers as well via an annotation.

Unless I'm missing something here, I still don't think you want to be inspecting arguments to determine how to encode and decode content. That's literally what the headers are for.